Kitodo/src/main/java/org/kitodo/production/services/schema/SchemaService.java

```diff
@@ -129,6 +132,29 @@ private void addVirtualFileGroupsToMetsMods(Workpiece workpiece, Process process
         }
     }
 
+    private void replaceFLocatForExport(Workpiece workpiece, Process process) throws URISyntaxException {
+        List<Folder> folders = process.getProject().getFolders();
+        for (MediaUnit mediaUnit : workpiece.getMediaUnits()) {
+            for (Entry<MediaVariant, URI> mediaFileForMediaVariant : mediaUnit.getMediaFiles().entrySet()) {
+                for (Folder folder : folders) {
+                    if (folder.getFileGroup().equals(mediaFileForMediaVariant.getKey().getUse())) {
+                        int lastSeparator = mediaFileForMediaVariant.getValue().toString().lastIndexOf(File.separator);
+                        String lastSegment = mediaFileForMediaVariant.getValue().toString()
+                                .substring(lastSeparator + 1);
+                        mediaFileForMediaVariant
+                                .setValue(new URI(replacePlaceholder(folder.getUrlStructure() + lastSegment, process)));
+                    }
+                }
+            }
+        }
+    }
+
+    private String replacePlaceholder(String uri, Process process) {
```

Comment:
Why you are not using the `replace` method from ` VariableReplacer` class? This method is doing the same.

Reply:
Thanks for noticing!